lib/tessel.js

```diff
@@ -441,6 +442,17 @@ function Tessel(options) {
     return new Pin(Object.assign({}, config, ToPortIdentity(index)));
   });
 
+  // REPL control:
+  // - If we detect that we're running on-board the Tessel, disable the repo
+  // - If the user specifies a repl option, use it no matter what
+  var tesselPathRegex = /^\/app\/remote-script/;
+  if (tesselPathRegex.test(__dirname)) {
```

Comment:
That function I mentioned above would replace __dirname in the suggestion I made here